contracts/polygon/connectors/hop/helpers.sol

```diff
@@ -0,0 +1,76 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+
+import { TokenInterface } from "../../common/interfaces.sol";
+import { DSMath } from "../../common/math.sol";
+import { Basic } from "../../common/basic.sol";
+import "./interface.sol";
+
+contract Helpers is DSMath, Basic {
+	/**
+	 * @param token The address of token to be bridged.(For USDC: 0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174)
+	 * @param chainId The Id of the destination chain.(For MAINNET : 1)
+	 * @param recipient The address to recieve the token on destination chain.
+	 * @param amount The total amount sent by user (Includes bonder fee, destination chain Tx cost).
+	 * @param bonderFee The fee to be recieved by bonder at destination chain.
+	 * @param amountOutMin minimum amount of token out for swap
+	 * @param deadline The deadline for the transaction (Recommended - Date.now() + 604800 (1 week))
+	 * @param destinationAmountOutMin minimum amount of token out for bridge, zero for L1 bridging
+	 * @param destinationDeadline The deadline for the transaction (Recommended - Date.now() + 604800 (1 week)), zero for L1 bridging
+	 */
+	struct BridgeParams {
+		address token;
+		uint256 chainId;
+		address recipient;
+		uint256 amount;
+		uint256 bonderFee;
```

Comment:
@pradyuman-verma Is there a way to get the bonderFee on-chain?

Reply:
`The fee should be set to the L1 gas price * 150,000 * 1.5 (The bonder will currently pick up transfers with underpriced fees but will likely get stricter in the future) + 0.18% of the amount.`

As per hop docs

Reply:
It might be easiest to just use the `getSendData()` function in the Hop SDK to perform this calculation in order give your users the optimal price that the bonder will accpept. The Hop frontend use it [here](https://github.com/hop-protocol/hop/blob/develop/packages/frontend/src/pages/Send/useSendData.tsx#L29).

Reply:
Ohh, thanks, this helps a lot.